float32 vel
string direction
int32 flag